### ISSUE TITLE
Suggestion for an alternative folder where to save history

### ISSUE THREAD

**Issue by seth100** [zabel-xyz/local-history#129]
Hi,

I would like to use this extension, but I don't want to have another folder inside my workspace / project.
I migrated to VSCode from Netbeans so I was used to have file history saved inside the IDE folder (`~/.netbeans/VERSION/var/filehistory/` on Ubuntu, if I am not wrong).
I want to use this extension in a similar way, could you please suggest a location on the system to put the history of all my workspaces? 
E.g. is it better to put it inside the extension folder: `~/.vscode-oss/extensions/xyz.local-history-1.8.1` or somewhere inside the VSCode config folder: `~/.config/VSCode/???`?

Thanks

**Comment by robross0606** [comment]
I already requested this (#110).  There are now several requests all about the same problem.  This should NOT be stored in the root of the workspace.  It borks so many things (linting, testing, searching, git commits, etc.)

**Comment by seth100** [comment]
@robross0606 I saw the issues and I get your point!
In my case it's fine to change the default behaviour on my own, but I would like to put local data inside a VSCode's folder instead of something related to the single project.
I tried to take a look at https://code.visualstudio.com/docs/editor/variables-reference but I'm not sure, that's why I wanted a suggestion from someone more expert than me.
As I said, IMHO a behaviour similar to other IDEs like Netbeans would be totally better, so storing local history inside a folder related to the application itself is a good idea.

**Comment by robross0606** [comment]
@seth100, I totally agree.  If you look at my ticket, I'm suggesting it should go in a subfolder in the user's home directory.  But I'd settle for `.vscode` which is where I put it right now by having to manually configure it all the time.  

If you look back thru the ticket history here, you'll see a TON of tickets all pertaining to this problem (#43, #47, #101, etc.).  The author simply refused to move the default even though it is in an ill-advised location.

**Comment by seth100** [comment]
@robross0606 yeah that's a good solution, but for example in my case (Linux and VSCode installed via Flatpak) the currently available alternatives are `~/.vscode-oss` or `~/.var/app/com.visualstudio.code/` or `~/.config/VSCode/` but at this point maybe the best way is to manually create `~/.vscode` and configure the extension to work in there!

**Comment by hervelemeur** [comment]
@robross0606 FYI you can configure it globally:

![](https://user-images.githubusercontent.com/52448671/104807355-c1295600-57de-11eb-9d9a-21522447d80c.png)

**Comment by mawg** [comment]
>     "local-history.path":     // Specify another location for .history folder (null: use workspaceFolder)
>         This settings must be an absolute path.

Even that doesn't work for me.

Let's day that I have 3 HS projects in

C:\localhost\htdocs\A
C:\localhost\htdocs\B
C:\localhost\htdocs\C

Each of these will have an idex.html file. When comparing local history, I obviously only want to check the history of that exact file, so naming them index_20151212205930.html, etc is unlikely to help.

Personally, when I ctrl+shift+F to search every file in the project, I do NOT want to search the history files. I appreciate that others may wish to do so, that's just how I prefer it. If I want to search the history, I will do it outside of the IDE - unless this plugin adds a "search in history" option.

For me, the ideal would be to specify a base path to backups, e.g c:\vsc_backups, and a variable part, which is the project folder name (only, not full path), giving me
c:\vsc_backups\A
c:\vsc_backups\B
c:\vsc_backups\C

so, c:\vsc_backups\A\index_20151212205930.html, etc

Does anyone agree?

**Comment by ganomi46** [comment]
I would like to also see the history in different directory, because it messes up with search, git, etc.
